rtlib/rope.hh

```diff
@@ -754,6 +756,49 @@ inline void append(rope::Ref<X> &str, const T &x) {
   str.append(x);
 }
 
+/* accepts a string and escapes all containing characters to be compatible
+ * to LaTeX math mode. Used for tikZ generations.
+ */
+inline std::string latex(const std::string in) {
+  std::string out = std::string(in);
+
+  /* https://www.cespedes.org/blog/85/how-to-escape-latex-special-characters
+   * note that we are in math mode, which might use different rules than
+   * LaTeX's "normal" text mode. */
+  boost::replace_all(out, "\\", "\\backslash");
+  boost::replace_all(out, "#", "\\#");
+  boost::replace_all(out, "$", "\\$");
+  boost::replace_all(out, "%", "\\%");
+  boost::replace_all(out, "&", "\\&");
+  boost::replace_all(out, "_", "\\_");
+  boost::replace_all(out, "{", "\\{");
+  boost::replace_all(out, "}", "\\}");
+  boost::replace_all(out, "^", "\\hat{\\ }");
```

Comment:
as I just learned the symbol `^` is called `caret` instead of hat :D

Reply:
hm, it looks like TeX sticks to hat and does not render properly with `\caret`. Who is wrong ;-)